Show default profiles in homematic cloud climate entity

hahn-th opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

Showing heating profiles in climate entity, if they have no custom name in the Homematicip App. If they have a custom name, they are already displayed.
Profiles are shown with following names:
PROFILE_1 = Default
PROFILE_2 = Alternative 1
PROFILE_3 = Alternative 2

This fixes issue #101455

Following your request, i have added myself as CodeOwner to manifest file
https://github.com/home-assistant/core/pull/99387


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #101455
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

hahn-th committed:
Show Default Profile Modes in Climate Entity

hahn-th committed:
Fix #101455 Change displayed names for default profiles

hahn-th committed:
Fix testcase

davet2001 commented:
@hahn-th Please can you update the PR title to describe the effect of the proposed change? Thanks!

hahn-th committed:
Fix missing test coverage

chbndrhnns commented:
What’s needed to proceed here?

davet2001 commented:
@chbndrhnns Merry Christmas!
Would you be willing to add your review to the PR?

chbndrhnns reviewed (approved):
I was running the branch locally and inspecting the changes in the UI. The changes fix the problem for me.

chbndrhnns commented:
@davet2001 I tried the changes from the branch on my local setup and it's fine. I think for the approval step here on Github, someone else needs to step in.

davet2001 commented:
@chbndrhnns A member approval is needed before it can be merged. I can do it, but the review needs to be more than just checking that it works.

Have you examined the diff? What did you think? Are you happy with all lines?

bdraco committed:
Bump roombapy to 1.6.10 (#106678)

changelog: https://github.com/pschmitt/roombapy/compare/1.6.8...1.6.10

fixes #105323

bdraco committed:
Fix missed cached_property for hvac_mode in climate (#106692)

kbickar committed:
Bump asyncsleepiq to v1.4.1 (#106682)

Update asyncsleepiq to v1.4.1

emontnemery committed:
Fix changed_variables in automation traces (#106665)

bdraco committed:
Bump thermobeacon-ble to 0.6.2 (#106676)

changelog: https://github.com/Bluetooth-Devices/thermobeacon-ble/compare/v0.6.0...v0.6.2

bdraco committed:
Pin lxml to 4.9.4 (#106694)

frenck committed:
Add missing vacuum toggle service description (#106729)

starkillerOG committed:
Bump reolink_aio to 0.8.5 (#106747)

bdraco committed:
Bump pyunifiprotect to 4.22.4 (#106749)

changelog: https://github.com/AngellusMortis/pyunifiprotect/compare/v4.22.3...v4.22.4

cgtobi committed:
Bump pyatmo to v8.0.2 (#106758)

puddly committed:
Bump ZHA dependencies (#106756)

* Bump ZHA dependencies

* Revert "Remove bellows thread, as it has been removed upstream"

This reverts commit c28053f4bf2539eb6150d35af19687610aaeac5e.

bdraco committed:
Bump habluetooth to 2.0.1 (#106750)

fixes switching scanners to quickly since the manager failed
to account for jitter in the auto discovered advertising interval

replaces and closes #96531

changelog: https://github.com/Bluetooth-Devices/habluetooth/compare/v2.0.0...v2.0.1

dmulcahey committed:
Fix Zlinky energy polling in ZHA (#106738)

frenck committed:
Bump version to 2024.1.0b3

dknowles2 committed:
Bump pyschlage to 2023.12.1 (#106782)

dknowles2 committed:
Bump pyschlage to 2023.12.1 (#106782)

Commit:
Bump pyunifiprotect to v4.22.5 (#106781)

Commit:
Bump pyunifiprotect to v4.22.5 (#106781)

cdce8p committed:
Update pytest to 7.4.4 (#106802)

bdraco committed:
Use shorthand attrs for tplink color temp min/max (#106796)

The valid_temperature_range property does a regex match over
every possible model. Avoid calling it more than once since
it will never change as its based on the model

Xitee1 committed:
Use walrus operator for roomba total cleaned area sensor value (#106772)

cdce8p committed:
Enable strict typing for airthings_ble (#106815)

cdce8p committed:
Enable strict typing for aprs (#106824)

cdce8p committed:
Enable strict typing for anel_pwrctrl (#106821)

cdce8p committed:
Enable strict typing for androidtv (#106820)

cdce8p committed:
Enable strict typing for alpha_vantage (#106816)

cdce8p committed:
Enable strict typing for aquostv (#106836)

cdce8p committed:
Enable strict typing for aruba (#106839)

cdce8p committed:
Enable strict typing for airq (#106813)

cdce8p committed:
Enable strict typing for amberelectric (#106817)

cdce8p committed:
Enable strict typing for asterisk_cdr + asterisk_mbox (#106841)

cdce8p committed:
Enable strict typing for arris_tg2492lg (#106838)

cdce8p committed:
Enable strict typing for arcam_fmj (#106837)

cdce8p committed:
Improve acmeda typing (#106812)

bdraco committed:
Bump bleak-retry-connector to 3.4.0 (#106831)

bdraco committed:
Bump bleak-retry-connector to 3.4.0 (#106831)

cereal2nd committed:
Bump pyduotecno to 2024.1.1 (#106801)

* Bump pyduotecno to 2024.0.1

* Bump pyduotecno to 2024.1.0

* small update

cereal2nd committed:
Bump pyduotecno to 2024.1.1 (#106801)

* Bump pyduotecno to 2024.0.1

* Bump pyduotecno to 2024.1.0

* small update

mdz committed:
Map missing preset mapping for heat mode "ready" in smarttub (#106856)

mdz committed:
Map missing preset mapping for heat mode "ready" in smarttub (#106856)

Lash-L committed:
Constrain dacite to at least 1.7.0 (#105709)

Lash-L committed:
Constrain dacite to at least 1.7.0 (#105709)

cdce8p committed:
Enable strict typing for ambiclimate (#106819)

bramkragten committed:
Update frontend to 20240101.0 (#106808)

bramkragten committed:
Update frontend to 20240101.0 (#106808)

bdraco committed:
Reduce duplicate code in json_loads (#106859)

tkdrob committed:
Set entity category and device class for Netgear LTE entities (#106661)

* Set entity category and device class for Netgear

* add suggested unit of measure and precision

kilrogg committed:
Fix fints account type check (#106082)

kilrogg committed:
Fix fints account type check (#106082)

zweckj committed:
Add diagnostics for tedee (#106662)

* add diagnostics

* don't redact lock name

* Update test_diagnostics.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

emontnemery committed:
Don't use entity_id in __repr__ of not added entity (#106861)

emontnemery committed:
Don't use entity_id in __repr__ of not added entity (#106861)

bdraco committed:
Bump bleak-esphome to 0.4.1 (#106832)

bdraco committed:
Fix incorrect state in Yale Access Bluetooth when lock status is unknown (#106851)

bdraco committed:
Fix incorrect state in Yale Access Bluetooth when lock status is unknown (#106851)

cdce8p committed:
Enable strict typing for airthings (#106814)

bdraco committed:
Bump pySwitchbot to 0.43.0 (#106833)

bdraco committed:
Bump pySwitchbot to 0.43.0 (#106833)

bdraco committed:
Bump yalexs-ble to 2.4.0 (#106834)

bdraco committed:
Bump yalexs-ble to 2.4.0 (#106834)

vexofp committed:
Pass default SSLContext instances to Octoprint custom HTTP sessions (#105351)

bdraco committed:
Use identity checks for HassJobType (#106860)

dknowles2 committed:
Fix Hydrawise data not refreshing (#105923)

Co-authored-by: Robert Resch <robert@resch.dev>

dknowles2 committed:
Fix Hydrawise data not refreshing (#105923)

Co-authored-by: Robert Resch <robert@resch.dev>

angristan committed:
Xiaomi MIIO: fix typo in error log (#106852)

bdraco committed:
Refactor restore state saving to avoid a dict lookup of ATTR_RESTORED (#106854)

bdraco committed:
Use shorthand attr for screenlogic climate preset modes (#106858)

bdraco committed:
Refactor light platform to avoid duplicate property calls (#106857)

bdraco committed:
Fix emulated_hue brightness check (#106783)

bdraco committed:
Fix emulated_hue brightness check (#106783)

jpbede committed:
Migrate vizio tests to use freezegun (#105417)

jpbede committed:
Migrate geo_rss_events test to use freezegun (#105895)

frenck committed:
Bump version to 2024.1.0b4

cdce8p committed:
Enable strict typing for arwn (#106840)

joostlek committed:
Improve entity descriptions in Tami4 (#106776)

* Improve entity descriptions in Tami4

* Improve entity descriptions in Tami4

* Improve entity descriptions in Tami4

* Improve entity descriptions in Tami4

* Improve entity descriptions in Tami4

iamrgroot committed:
Changed setup of EnergyZero services (#106224)

* Changed setup of energyzero services

* PR review updates

* Dict access instead of get

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Added tests for unloaded state

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

iamrgroot committed:
Changed setup of EnergyZero services (#106224)

* Changed setup of energyzero services

* PR review updates

* Dict access instead of get

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Added tests for unloaded state

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

zweckj committed:
Add reauthentication for tedee integration (#106667)

* start work

* add reauth

* title

* reuse user step for reauth

* Update strings.json

* simplify flow

* remove inline if

* remove await hass block

jpbede committed:
Avoid unnecessary domain dataclass in Discovergy (#106869)

frenck committed:
Mark humidifier entity component as strictly typed (#106721)

Co-authored-by: Marc Mueller <30130371+cdce8p@users.noreply.github.com>

cdce8p committed:
Enable strict typing for apache_kafka (#106823)

steffenrapp committed:
Add translatable title to logbook (#106810)

autinerd committed:
Move urllib3 constraint to pyproject.toml (#106768)

autinerd committed:
Move urllib3 constraint to pyproject.toml (#106768)

mib1185 committed:
Apply late review comments on media player (#106727)

mib1185 committed:
Apply late review comments on media player (#106727)

emontnemery committed:
Enable strict typing of date_time (#106868)

* Enable strict typing of date_time

* Fix parse_datetime

* Add test

* Add comments

* Update tests/util/test_dt.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

emontnemery committed:
Deprecate 'beat' display option in Time & Date (#106871)

* Deprecate 'beat' display option in Time & Date

* Move deprecation warning

* Update homeassistant/components/time_date/const.py

Co-authored-by: Sander <developer@golles.nl>

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>
Co-authored-by: Sander <developer@golles.nl>

dmulcahey committed:
Bump Zigpy to 0.60.4 (#106870)

dmulcahey committed:
Bump Zigpy to 0.60.4 (#106870)

bdraco committed:
Index entities by domain for entity services (#106759)

starkillerOG committed:
Add Reolink image settings (#105415)

emontnemery committed:
Improve time_date tests (#106878)

emontnemery committed:
Fix state update in time_date sensor (#106879)

emontnemery committed:
Handle time zone change in time_date (#106880)

emontnemery committed:
Disable polling in time_date sensor (#106881)

frenck committed:
Mark stt entity component as strictly typed (#106723)

Co-authored-by: Marc Mueller <30130371+cdce8p@users.noreply.github.com>

pnbruckner committed:
Remove deprecated Life360 yaml configuration (#106286)

bdraco committed:
Close stale connections in yalexs_ble to ensure setup can proceed (#106842)

bdraco committed:
Close stale connections in yalexs_ble to ensure setup can proceed (#106842)

allenporter committed:
Improve fitbit authentication error handling (#106885)

allenporter committed:
Improve fitbit authentication error handling (#106885)

bdraco committed:
Replace intersection with isdisjoint in apple_tv config flow (#106633)

allenporter committed:
Improve To-do service error handling (#106886)

allenporter committed:
Improve To-do service error handling (#106886)

cdce8p committed:
Add strict typing to command_line (#106889)

* Add strict typing to command_line

* Code review

cdce8p committed:
Enable strict typing for co2signal (#106888)

bdraco committed:
Avoid tuple construction to check HKC available (#106902)

bramkragten committed:
Update frontend to 20240102.0 (#106898)

bramkragten committed:
Update frontend to 20240102.0 (#106898)

bdraco committed:
Update switchbot to use close_stale_connections_by_address (#106835)

bdraco committed:
Update switchbot to use close_stale_connections_by_address (#106835)

cdce8p committed:
Enable strict typing for blueprint (#106887)

bdraco committed:
Bump aiohttp-zlib-ng to 0.2.0 (#106691)

cdce8p committed:
Enable strict typing for axis (#106844)

frenck committed:
Bump version to 2024.1.0b5

cdce8p committed:
Remove assert for unique_id (#106910)

* Remove assert for unique_id

* Use str | None return instead

synesthesiam committed:
Bump intents to 2024.1.2 (#106909)

synesthesiam committed:
Bump intents to 2024.1.2 (#106909)

zweckj committed:
Add tedee bridge as via_device for tedee integration (#106914)

* add bridge as via_device

* add bridge as via_device

* move getting bridge to update_data

* add bridge property

steffenrapp committed:
Add translatable title to holiday (#106825)

mib1185 committed:
Remove group_members from significant attributes in media player (#106916)

mib1185 committed:
Remove group_members from significant attributes in media player (#106916)

finder39 committed:
Fix qBittorrent torrent count when empty (#106903)

* Fix qbittorrent torrent cound when empty

* lint fix

* Change based on comment

jpbede committed:
Avoid triggering ping device tracker `home` after restore (#106913)

jpbede committed:
Avoid triggering ping device tracker `home` after restore (#106913)

jpbede committed:
Bump aioelectricitymaps to v0.1.6 (#106932)

cdce8p committed:
Enable strict typing for counter (#106906)

bdraco committed:
Bump SQLAlchemy to 2.0.25 (#106931)

* Bump SQLAlchemy to 2.0.25

changelog: https://docs.sqlalchemy.org/en/20/changelog/changelog_20.html#change-2.0.25

* drop unused ignore now that upstream is fixed

bdraco committed:
Speed up getting the mean of statistics (#106930)

All the values we need to get the mean for are always list[float]
so we can use a much simpler algorithm to get the mean of the list

erwindouna committed:
Add try-catch for invalid auth to Tado (#106774)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

erwindouna committed:
Add try-catch for invalid auth to Tado (#106774)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

frenck committed:
Bump version to 2024.1.0b6

erwindouna committed:
Change Tado deprecation version to 2024.7.0 (#106938)

Change version to 2024.7.0

erwindouna committed:
Change Tado deprecation version to 2024.7.0 (#106938)

Change version to 2024.7.0

bramkragten committed:
Update frontend to 20240103.0 (#106942)

bramkragten committed:
Update frontend to 20240103.0 (#106942)

frenck committed:
Bump version to 2024.1.0b7

edenhaus committed:
Fix creating cloud hook twice for mobile_app (#106945)

edenhaus committed:
Fix creating cloud hook twice for mobile_app (#106945)

bramkragten committed:
Update frontend to 20240103.1 (#106948)

bramkragten committed:
Update frontend to 20240103.1 (#106948)

janiversen committed:
Only set precision in modbus if not configured. (#106952)

Only set precision if not configured.

janiversen committed:
Only set precision in modbus if not configured. (#106952)

Only set precision if not configured.

crug80 committed:
Removed double assignment of _attr_target_temperature_step in __init__ (#106611)

Removed double/wrong assignement of _attr_target_temperature_step into climate Init routine

chbndrhnns commented:
@hahn-th I think the feature branch needs to be rebased first, I saw a conflict while trying that on my machine. Could you give it a try?

lellky committed:
Set precision to halves in flexit_bacnet (#106959)

flexit_bacnet: set precision to halves for target temperature

lellky committed:
Set precision to halves in flexit_bacnet (#106959)

flexit_bacnet: set precision to halves for target temperature

bramkragten committed:
Update frontend to 20240103.3 (#106963)

bramkragten committed:
Update frontend to 20240103.3 (#106963)

frenck committed:
Bump version to 2024.1.0b8

frenck committed:
Bump version to 2024.1.0

mib1185 committed:
Add support for placeholders in entity name translations (#104453)

* add placeholder support to entity name translation

* add negativ tests

* make property also available via description

* fix doc string in translation_placeholders()

* fix detection of placeholder

* validate placeholders for localized strings

* add test

* Cache translation_placeholders property

* Make translation_placeholders uncondotionally return dict

* Fall back to unsubstituted name in case of mismatch

* Only replace failing translations with English

* Update snapshots

* Blow up on non stable releases

* Fix test

* Update entity.py

---------

Co-authored-by: Erik <erik@montnemery.com>
Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

frenck committed:
2024.1.0 (#106970)

LaStrada committed:
Close stale connections (Airthings BLE) (#106748)

Co-authored-by: J. Nick Koston <nick@koston.org>

pfrazer committed:
Bump dropmqttapi to 1.0.2 (#106978)

rfverbruggen committed:
Bump rachiopy to 1.1.0 (#106975)

autinerd committed:
Bump openwebifpy to 4.0.4 (#107000)

bdraco committed:
Fix missing backwards compatiblity layer for humidifier supported_features (#107026)

fixes #107018

bieniu committed:
Get Shelly RPC device `gen` from config entry data (#107019)

Use gen from config entry data

bdraco committed:
Fix ESPHome service removal when the device name contains a dash (#107015)

* Fix ESPHome service removal when the device name contains a dash

If the device name contains a dash the service name is mutated to
replace the dash with an underscore, but the remove function did
not do the same mutation so it would fail to remove the service

* add more coverage

* more cover

teharris1 committed:
Bump pyinsteon (#107010)

bdraco committed:
Small cleanups to ESPHome light platform (#107003)

- Remove unreachable code
- Cache filtering when possible
- Add missing coverage

bdraco committed:
Add missing coverage for esphome_state_property decorator (#106998)

bdraco committed:
Fix first ESPHome device update entity not offering install feature (#106993)

In the case where the user gets their first ESPHome device such as a RATGDO,
they will usually add the device first in HA, and than find the dashboard.

The install function will be missing because we do not know if the dashboard
supports updating devices until the first device is added. We now set the
supported features when we learn the version when the first device is added

jpbede committed:
Use call_soon_threadsafe in token updater of Ring (#106984)

Use call_soon_threadsafe in token update of Ring

balloob committed:
Ring: Add partial back (#107040)

engrbm87 committed:
Issue warning if glances server version is 2 (#105887)

* Issue warning if glances server version is 2

* Auto detect api version

* Apply suggestions

* Add HA version deprecation

* Apply suggestions from code review

* update config flow tests

* Fix breaks in ha version

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bachya committed:
Re-architect Guardian to use better entity descriptions and properties (#107028)

* Re-architect Guardian to use better entity descriptions and properties

* Reduce blast area

* Code review

* Remove mixins

jkrall committed:
Add button platform to Opengarage (#103569)

* Add button entity to reboot OpenGarage device

* Addressing code review comments

* Another code-review fix

* Update homeassistant/components/opengarage/button.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

joostlek committed:
Use async_register in streamlabswater (#107060)

joostlek committed:
Fix data access in streamlabs water (#107062)

* Fix data access in streamlabs water

* Fix data access in streamlabs water

bdraco committed:
Small cleanups to denonavr (#107050)

thomaskistler committed:
Update pydrawise to 2024.1.0 (#107032)

bdraco committed:
Add test coverage for ESPHome service calls (#107042)

rappenze committed:
Remove dead code in fibaro light (#106890)

joostlek committed:
Add sensor tests to Streamlabs water (#107065)

frenck committed:
Update home-assistant/builder to 2024.01.0 (#107069)

jbouwh committed:
Refactor drop sensor tests (#106965)

* Refactor drop sensor tests

* Setup the config entry instead of the component

joostlek committed:
Add entity descriptions to Streamlabs water (#107071)

* Add sensor tests to Streamlabs water

* Add sensor tests to Streamlabs water

* Use entity descriptions in streamlabs water

* Use entity descriptions in streamlabs water

* Use entity descriptions in streamlabs water

* Add translations

jbouwh committed:
Refactor drop tests for binary_sensor (#107090)

bdraco committed:
Fix missing backwards compatibility layer for water_heater supported_features (#107091)

emontnemery committed:
Deduplicate handling of duplicated constants (#107074)

* Deduplicate handling of duplicated constants

* Use DeprecatedConstant + DeprecatedConstantEnum

* Fixup

* Remove test cases with unnamed tuples

frenck committed:
Merge branch 'master' into dev

bdraco committed:
Bump habluetooth to 2.0.2 (#107097)

MartinHjelmare committed:
Remove zwave_js numeric sensor rounding (#107100)

bdraco committed:
Revert "Bump aiohttp-zlib-ng to 0.2.0 (#106691)" (#107109)

jbouwh committed:
Rework drop_connect switch, select and coordinator tests and cleanup fixtures (#107119)

* Refactor drop_connect switch and select tests

* Update coordinator tests, cleanup fixtures

bdraco committed:
Move group helpers into their own module (#106924)

This gets rid of the legacy need to use bind_hass, and
the expand function no longer looses typing.

jbouwh committed:
Reorganize drop_connect tests (#107148)

bramkragten committed:
Update frontend to 20240104.0 (#107155)

emericklaw committed:
Bump Orvibo to 1.1.2 (#107162)

* Bump python-orvibo version

Fixes https://github.com/home-assistant/core/issues/106923

* Add version number

* Remove version

* Bump python-orvibo version

joostlek committed:
Migrate Emonitor to has entity name (#107153)

jrieger committed:
Migrate AVM FRITZ!Box Call monitor to has entity name (#99752)

* Migrate AVM FRITZ!Box Call monitor to has entity name

* Update homeassistant/components/fritzbox_callmonitor/strings.json

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update homeassistant/components/fritzbox_callmonitor/strings.json

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update sensor.py

* Update sensor.py

* Update strings.json

* Use translation placeholders

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

bdraco committed:
Cache homekit_controller supported features (#106702)

joostlek committed:
Use snapshots in Glances sensor tests (#107159)

* Use snapshots in Glances sensor tests

* yes

Luap99 committed:
Pass down language to hassil (#106490)

Hassil needs the language to convert numbers, this was added in
https://github.com/home-assistant/hassil/pull/78. This fixes an
annoying warning from the logs.

Fixes #104760

joostlek committed:
Remove precision in streamlabs water (#107096)

frenck committed:
Update Home Assistant base image to 2024.01.0 - Python 3.12 (#107175)

zweckj committed:
Pass aiohttp clientsession to tedee integration (#107089)

* pass aiohttpsession

* Update homeassistant/components/tedee/config_flow.py

Co-authored-by: Sid <27780930+autinerd@users.noreply.github.com>

* Update homeassistant/components/tedee/__init__.py

Co-authored-by: Sid <27780930+autinerd@users.noreply.github.com>

* move to coordinator

---------

Co-authored-by: Sid <27780930+autinerd@users.noreply.github.com>

Shulyaka committed:
Add conversation_id parameter to conversation.process service (#106078)

* Add conversation_id parameter to conversation.process service

* fix test

* fix tests

bachya committed:
Clean up outdated entity replacement logic in Guardian (#107160)

joostlek committed:
Introduce base entity in streamlabs water (#107095)

bdraco committed:
Bump aiohomekit to 3.1.2 (#107177)

ashleigh-hopkins committed:
Update sensorpush-ble library to 1.6.1 (#107168)

bdraco committed:
Fix tplink overloading power strips (#104208)

mib1185 committed:
Fix switch states in AVM FRITZ!Box Tools (#107183)

erwindouna committed:
Bump to PyTado 0.17.3 (#107181)

synesthesiam committed:
Report missing entities/areas instead of failing to match in Assist (#107151)

* Report missing entities/areas instead of failing

* Fix test

* Update assist pipeline test snapshots

* Test complete match failure

* Fix conflict

joostlek committed:
Fix conversation snapshots (#107196)

tronikos committed:
Disable IPv6 in the opower integration to fix AEP utilities (#107203)

bdraco committed:
Bump bluetooth-adapters to 0.17.0 (#107195)

changelog: https://github.com/Bluetooth-Devices/bluetooth-adapters/compare/v0.16.2...v0.17.0

related https://github.com/home-assistant/operating-system/issues/2944

MartinHjelmare committed:
Set zwave_js voltage sensor suggested precision (#107116)

bdraco committed:
Change default python version to 3.12 for image builds (#107209)

bdraco committed:
Attempt to fix 32bit docker builds (#107210)

Bre77 committed:
Hotfix cache logic bug in Tessie (#107187)

ppetru committed:
Fix entity property cache creation arguments (#107221)

jbouwh committed:
Remove work-a-round for mqtt sensors with an entity_category set to `config` (#107199)

* Remove work-a-round for mqtt sensors with an entity_category set to `config`

* Cleanup strings

jbouwh committed:
Remove unneeded preset_mode checks for mqtt climate (#107190)

jbouwh committed:
Remove naming warnings and work-a-rounds for incorrectly configured MQTT entities (#107188)

* Remove naming warnings for MQTT entities

* Remove unused const

bdraco committed:
Add color temp support for older HomeKit devices (#107206)

ludeeus committed:
Use supported_features_compat in update.install service (#107224)

edenhaus committed:
Fix mobile_app cloudhook creation (#107068)

emontnemery committed:
Include deprecated constants in wildcard imports (#107114)

joostlek committed:
Finish Efergy entity translations (#107152)

raman325 committed:
Bump zwave-js-server-python to 0.55.3 (#107225)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Use identity checks for unifiprotect enums (#106795)

enums are singletons in this case and there is no need to use
the slower equality checks here

CodeFatherG committed:
Add Tyua Product Category "dsd" for Filament Light (#106709)

* Add Product Category "dsd" support to tuya integration for Filament Lights

* remove unnecessary color_temp and color_data arguments

---------

Co-authored-by: Jan Čermák <sairon@users.noreply.github.com>

cereal2nd committed:
Bump velbus-aio to 2023.12.0 (#107066)

bdraco committed:
Add test coverage for ESPHome device info (#107034)

cottsay committed:
Add Rainforest RAVEn integration (#80061)

* Add Rainforest RAVEn integration

* Add Rainforest Automation brand

* Add diagnostics to Rainforest RAVEn integration

* Drop a test assertion for an undefined behavior

* Add DEVICE_NAME test constant

* Catch up with reality

* Use Platform.SENSOR

Co-authored-by: Robert Resch <robert@resch.dev>

* Make rainforest_raven translatable

* Stop setting device_class on unsupported scenarios

* Rename rainforest_raven.data -> rainforest_raven.coordinator

* Make _generate_unique_id more reusable

* Move device synchronization into third party library

* Switch from asyncio_timeout to asyncio.timeout

* Ignore non-electric meters

Co-authored-by: Robert Resch <robert@resch.dev>

* Drop direct dependency on iso4217, bump aioraven

* Use RAVEn-specific exceptions

* Add timeouts to data updates

* Move DeviceInfo generation from Sensor to Coordinator

* Store meter macs as strings

* Convert to using SelectSelector

* Drop test_flow_user_invalid_mac

This test isn't necessary now that SelectSelector is used.

* Implement PR feedback

- Split some long format lines
- Simplify meter mac_id extraction in diagnostics
- Expose unique_id using an attribute instead of a property
- Add a comment about the meters dictionary shallow copy

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Simplify mac address redaction

Co-authored-by: Joakim Sørensen <ludeeus@ludeeus.dev>

* Freeze RAVEnSensorEntityDescription dataclass

Co-authored-by: Erik Montnemery <erik@montnemery.com>

---------

Co-authored-by: Robert Resch <robert@resch.dev>
Co-authored-by: Erik Montnemery <erik@montnemery.com>
Co-authored-by: Joakim Sørensen <ludeeus@ludeeus.dev>

RoboMagus committed:
Add support for service response to RESTful command (#97208)

* Add ServiceResponse to rest_command

* Handle json and text responses.
Add Unit tests

* Rest command text output handling.
Prevent issue solved by PR#97777

* Re-raise exceptions as HomeAssistantError to enable 'continue_on_error' in scripts / automations.

* Improve test coverage

* Restructure to improve McCabe Complexity

* Remove LookupError

* Revert exception catching location

* Remove LookupError from exception handling

bdraco committed:
Bump aiohttp-zlib-ng to 0.3.0 (#107184)

rikroe committed:
Add python_script response (#97937)

* Add response to python_script

* Reset output to empty dict if not valid dict

* Add tests for python_script response

* Raise Exceptions on service execution

* Add info on exception type

* Raise ServiceValidationError instead of ValueError

* Raise only on return_response=True

* Fix exception logger if no service response

* Create issue if exception is not raised

* Revert "Create issue if exception is not raised"

This reverts commit a61dd8619fdf09585bbd70e1392d476f3fcea698.

---------

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

MisterCommand committed:
Add Hong Kong Observatory integration (#98703)

* Add Hong Kong Observatory integration

* Move coordinator to a separate file

* Map icons to conditions

* Fix code for review

* Skip name

* Add typings to data_coordinator

* Some small fixes

* Rename coordinator.py

jpbede committed:
Make exceptions in rest_command services translatable (#107252)

al-s committed:
Add new Rabbit Air integration (#66130)

* Add new Rabbit Air integration

* Remove py.typed file

It is not needed and was just accidentally added to the commit.

* Enable strict type checking for rabbitair component

Keeping the code fully type hinted is a good idea.

* Add missing type annotations

* Remove translation file

* Prevent data to be added to hass.data if refresh fails

* Reload the config entry when the options change

* Add missing type parameters for generics

* Avoid using assert in production code

* Move zeroconf to optional dependencies

* Remove unnecessary logging

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Remove unused keys from the manifest

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Replace property with attr

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Allow to return None for power

The type of the is_on property now allows this.

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Remove unnecessary method call

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Update the python library

The new version properly re-exports names from the package root.

* Remove options flow

Scan interval should not be part of integration configuration. This was
the only option, so the options flow can be fully removed.

* Replace properties with attrs

* Remove multiline ternary operator

* Use NamedTuple for hass.data

* Remove unused logger variable

* Move async_setup_entry up in the file

* Adjust debouncer settings to use request_refresh

* Prevent status updates during the cooldown period

* Move device polling code to the update coordinator

* Fix the problem with the switch jumping back and forth

The UI seems to have a timeout of 2 seconds somewhere, which is just a
little bit less than what we normally need to get an updated state. So
the power switch would jump to its previous state and then immediately
return to the new state.

* Update the python library

The new version fixes errors when multiple requests are executed
simultaneously.

* Fix incorrect check for pending call in debouncer

This caused the polling to stop.

* Fix tests

* Update .coveragerc to exclude new file.
* Remove test for Options Flow.

* Update the existing entry when device access details change

* Add Zeroconf discovery step

* Fix tests

The ZeroconfServiceInfo constructor now requires one more argument.

* Fix typing for CoordinatorEntity

* Fix signature of async_turn_on

* Fix depreciation warnings

* Fix manifest formatting

* Fix warning about debouncer typing

relates to 5ae5ae5392729b4c94a8004bd02e147d60227341

* Wait for config entry platform forwards

* Apply some of the suggested changes

* Do not put the MAC address in the title. Use a fixed title instead.
* Do not format the MAC to use as a unique ID.
* Do not catch exceptions in _async_update_data().
* Remove unused _entry field in the base entity class.
* Use the standard attribute self._attr_is_on to keep the power state.

* Store the MAC in the config entry data

* Change the order of except clauses

OSError is an ancestor class of TimeoutError, so TimeoutError should be
handled first

* Fix depreciation warnings

* Fix tests

The ZeroconfServiceInfo constructor arguments have changed.

* Fix DeviceInfo import

* Rename the method to make it clearer what it does

* Apply suggestions from code review

* Fix tests

* Change speed/mode logic to use is_on from the base class

* A zero value is more appropriate than None

since None means "unknown", but we actually know that the speed is zero
when the power is off.

---------

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Erik Montnemery <erik@montnemery.com>

frenck committed:
Update Home Assistant Wheels action to 2024.01.0 (#107240)

wilburCforce committed:
Add Lutron config flow (#98489)

* rough in structure for config_flow

* updated json files

* initial conversion to config_flow

* minor updates

* Update binary_sensor.py

* Update config_flow.py

* Update __init__.py

* updates beased on requested changes

* Update const.py

added doc note for ruff

* updated based on suggestions

* updated load xmldb for efficiency

* updated references

* removed unneeded file

* updated config flow to use GUID from XML DB

* minor update to change logging

* updated based on PR feedback

* reverted change for line 30 based on testing

* corrected user_input

* updated based on latest comments

* exception handling

* added raising of issues for config flow

* updated issues strings

* config flow test shell - needs work

* minor changes

* Update strings.json

* Update config_flow.py

* Update __init__.py

* Create conftest.py

* Update test_config_flow.py

* Update test_config_flow.py

* Update config_flow.py

* Update strings.json

* Update requirements_test_all.txt

* Update strings.json

* Update strings.json

* Update config_flow.py

* Update test_config_flow.py

* Update config_flow.py

* Create test_init.py

* Update __init__.py

* Delete tests/components/lutron/test_init.py

* Update strings.json

* updated import parts and tested

* updated strings to improve user feedback

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

SLaks committed:
Allow selecting of counter entities in derivative/integration config flow (#105321)

joostlek committed:
Sort Lutron platforms (#107257)

jpbede committed:
Remove duplicate assignment of `median` and `statistical_mode` jinja2 filter (#106953)

cdce8p committed:
Improve denonavr typing (#106907)

cdce8p committed:
Enable strict typing for downloader (#107263)

joostlek committed:
Migrate Suez Water to has entity name (#107251)

cdce8p committed:
Enable strict typing for minecraft_server (#107262)

hahn-th committed:
merge

janiversen commented:
You just requested from just about all active developers. Please rebase your pull request and try again..

hahn-th commented:
@janiversen uups i just tried to rebase and that was my result.... something went wrong i think
